src/content_container.rs

```diff
@@ -117,7 +122,141 @@ impl Serialize for NestedPart {
     }
 }
 
-#[derive(Debug, Clone)]
+impl<'de> Deserialize<'de> for NestedPart {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        struct NestedPartVisitor;
+        impl<'de> Visitor<'de> for NestedPartVisitor {
+            type Value = NestedPart;
+
+            fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+                formatter.write_str("struct NestedPart")
+            }
+
+            fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
+            where
+                A: serde::de::SeqAccess<'de>,
+            {
+                // Keep track of current array index for better error messages
+                let mut current_i = 0;
+                let mut i = || {
+                    let old = current_i;
+                    current_i += 1;
+                    old
+                };
+
+                let disposition = seq
+                    .next_element::<Disposition>()?
+                    .ok_or_else(|| de::Error::invalid_length(i(), &self))?;
```

Comment:
Since you're using this quite often, maybe a helper function would help here. Maybe something like

```rust
fn next_element<'a, A: serde::de::SeqAccess<'a>>(
    seq: &mut A,
    np: &NestedPartVisitor,
    i: &mut dyn FnMut() -> usize,
) -> Result<Disposition, A::Error> {
    Ok(seq
        .next_element::<Disposition>()?
        .ok_or_else(|| de::Error::invalid_length(i(), np))?)
}
```

Reply:
I'm rewriting the serde implementations with a custom macro now, because it will be reusable and more maintainable. I will finish and push the changes tomorrow.